### PR TITLE
refactor(console): config the allowed mime type for ImageUploader component

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

Many email services do not accept SVG. So to avoid this issue, we need to exclude SVG files when uploading images.

Here comes the change to fix the aforementioned issue:
enable the customization on accepting image format for `<ImageFileUploader />` component. And restrict the logo uploaded for email branding to be in `png`, `jpg` or `jpeg` format.

## Testing

Tested locally, can not upload SVG image and the image with space in its file name can successfully show up in email content.
1. can not upload SVG file:
image

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [ ] docs

OR

- [x] This PR is not applicable for the checklist

### CODE REVIEW

File: packages/console/src/ds-components/Uploader/ImageUploader/index.tsx [hunk 1]
```diff
@@ -48,7 +50,15 @@ function ImageUploader({ name, value, onDelete, ...rest }: Props) {
       </IconButton>
     </div>
   ) : (
-    <FileUploader allowedMimeTypes={allowedImageMimeTypes} maxSize={maxImageSizeLimit} {...rest} />
+    <FileUploader
+      allowedMimeTypes={
+        // Fallback to default allowed image mime types if `allowedMimeTypes` is not provided.
+        conditional((allowedMimeTypes ?? []).length > 0 && allowedMimeTypes) ??
```
Comment (gao-sun) [inline packages/console/src/ds-components/Uploader/ImageUploader/index.tsx:56]:
```suggestion
        Boolean(allowedMimeTypes?.length) ? allowedImageMimeTypes : allowedMimeTypes
```

File: packages/console/src/ds-components/Uploader/ImageUploader/index.tsx [hunk 2]
```diff
@@ -48,7 +50,15 @@ function ImageUploader({ name, value, onDelete, ...rest }: Props) {
...
+        allowedImageMimeTypes
```
Comment (gao-sun) [inline packages/console/src/ds-components/Uploader/ImageUploader/index.tsx:57]:
```suggestion
        Boolean(allowedMimeTypes?.length) ? allowedImageMimeTypes : allowedMimeTypes
```

File: packages/console/src/ds-components/Uploader/ImageUploaderField/index.tsx [hunk 3]
```diff
@@ -11,21 +12,23 @@ import * as styles from './index.module.scss';
...
 };
 
-function ImageUploaderField({ onChange, ...rest }: Props) {
+function ImageUploaderField({ onChange, allowedMimeTypes, ...rest }: Props) {
   const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
 
   const limitDescription = t('components.uploader.image_limit', {
     size: maxImageSizeLimit / 1024,
-    extensions: convertToFileExtensionArray(allowedImageMimeTypes),
+    extensions: convertToFileExtensionArray(allowedMimeTypes ?? allowedImageMimeTypes),
```
Comment (gao-sun) [inline packages/console/src/ds-components/Uploader/ImageUploaderField/index.tsx:23]: no need to fall back everywhere? just do it once

File: packages/core/src/utils/storage/utils.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { conditionalString } from '@silverhand/essentials';
+
+export const fileNameURLEncoder = (url: string): string => {
+  const urlSplitter = url.split('/');
+  const filePath = urlSplitter.slice(0, -1).join('/');
+  const fileName = urlSplitter.at(-1);
+  return `${filePath}${conditionalString(fileName && '/')}${conditionalString(
+    fileName && encodeURIComponent(fileName)
+  )}`;
```
Comment (gao-sun) [inline packages/core/src/utils/storage/utils.ts:9]:
```suggestion
  const splitter = fullPath.split('/');
  const filename = splitted.at(-1);

  return filename ? [...splitter, encodeURIComponent(filename)].join('/') : fullPath;
```

File: packages/core/src/utils/storage/utils.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { conditionalString } from '@silverhand/essentials';
+
+export const fileNameURLEncoder = (url: string): string => {
```
Comment (gao-sun) [inline packages/core/src/utils/storage/utils.ts:3]:
```suggestion
export const fileNameUrlEncoder = (url: string): string => {
```

Comment (gao-sun) [reply packages/core/src/utils/storage/utils.ts:3]: also we should add unit tests for this

**Review by gao-sun — APPROVED** [review]

File: packages/core/src/routes/user-assets.ts [hunk 6]
```diff
@@ -76,13 +77,13 @@ export default function userAssetsRoutes<T extends AuthedRouter>(...[router]: Ro
...
       try {
-        const { url } = await uploadFile(await readFile(file.filepath), objectKey, {
+        const { url: rawUrl } = await uploadFile(await readFile(file.filepath), objectKey, {
           contentType: file.mimetype,
           publicUrl: storageProviderConfig.publicUrl,
         });
 
         const result: UserAssets = {
-          url,
+          url: fileNameUrlEncoder(rawUrl),
```
Comment (wangsijie) [inline packages/core/src/routes/user-assets.ts:86]: I think we should do the encode in Console, the response url is already valid?

**Review by wangsijie — CHANGES_REQUESTED** [review]

File: packages/connectors/connector-logto-email/src/utils.ts [hunk 7]
```diff
@@ -0,0 +1,5 @@
+export const fileNameUrlEncoder = (url: string): string => {
```
Comment (wangsijie) [inline packages/connectors/connector-logto-email/src/utils.ts:1]: use `encodeURI`?

**Review by wangsijie — APPROVED** [review]

**Pull request merged**